oolab/src/main/java/agh/ics/oop/GrassField.java

```diff
@@ -0,0 +1,80 @@
+package agh.ics.oop;
+
+import org.w3c.dom.html.HTMLMapElement;
+
+import java.lang.reflect.Array;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+
+public class GrassField extends AbstractWorldMap{
+
+    private int fieldsNum;
+    private Vector2d[] positionsVect;
+
+    //private List<Grass> grasses = new ArrayList<>();
+
+    public GrassField(int fieldsNum) {
+        this.fieldsNum = fieldsNum;
+        int fieldsRoot = (int)Math.sqrt(10*(double)fieldsNum);
+        System.out.println(fieldsRoot);
+
+        List<Integer> positions = IntStream.range(0, fieldsRoot*fieldsRoot)
+                .boxed()
+                .collect(Collectors.toList());
+        System.out.println(positions);
+        Collections.shuffle(positions);
+        System.out.println(positions);
+        int[] array = positions.stream().mapToInt(i -> i).toArray();
+        Object[] positionsArray =  Arrays.stream(array)
+                .limit(fieldsNum)
+                .mapToObj(x -> new Vector2d(x/fieldsRoot, x%fieldsRoot))
+                .toArray();
+        positionsVect = new Vector2d[positionsArray.length];
```

Comment:
rozwiązanie fajne i deterministyczne, tylko przydałoby się to jeszcze uporządkować, najlepiej podzielić na metody (ogólnie w konstruktorze nie powinno byc za dużo "nienazwanej" logiki)